### PR TITLE
feat: Make e2e with devspace binary sync

### PR DESCRIPTION
**Pull request by marekfexa-outreach**
## What this PR does / why we need it
- Adds an alternative mode for `USE_DEVSPACE=true make e2e` that allows to run tests inside of k8s cluster. It uses devspace binary sync.
- Support run in CircleCI - here is the way how to early adopt - https://github.com/getoutreach/devenvtestapp/pull/8
- Supports junit reporting 

## Jira ID

[QF-880]

## Notes for your reviewers

[QF-880]: https://outreach-io.atlassian.net/browse/QF-880?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Erik-Outreach** [comment]
I noticed e2e job still creates a `devenv tunnel`, is it still needed?

**Review by Erik-Outreach — APPROVED** [review]
Just one idea about moving part of the logic to devenv to make easier to test, otherwise I don't see any blockers

File: e2e/e2e.go [hunk 1]
```diff
@@ -219,6 +223,84 @@ func shouldRunE2ETests() (bool, error) {
...
+	}
+	testsSuccess, err := parseResultFromJunitReport()
+	if err != nil {
+		return err
+	}
+	if !testsSuccess {
+		return errors.New("E2E Tests failed")
+	}
+	log.Info().Msg("E2E Tests succeeded.")
+	return nil
```
[the remark this answers is not in the document]
Comment (marekfexa-outreach) [reply e2e/e2e.go:280]: Yeah, I agree, that seems like a good idea

[the remark this answers is not in the document]
Comment (Erik-Outreach) [reply e2e/e2e.go:280]: Should we create a Jira just in case?

**Pull request merged**